modules/power/sleep.nix

```diff
@@ -0,0 +1,80 @@
+{ config, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.power.sleep;
+
+  onOff = cond: if cond then "on" else "off";
+in
+
+{
+  options = {
+    power.sleep.computer = mkOption {
+      type = types.nullOr (types.either types.ints.unsigned (types.enum ["never"]));
```

Comment:
If `0` and `never` behave identically, I think the standard way that is handled in nix-darwin is not to accept a "never" option and to just note in the description that `0` disables the feature.

Reply:
I can't remember my original reasoning for writing the module the way I did, but I would much rather disable sleep by specifying `never` than having to remember what `0` means or having to consult the documentation; `never` is more readable.

The `systemsetup` man page describes to use `never` or `off` to disable the sleeping, but does not specify what happens if `0` is set. That they behave identically have been derived empirically.

If it's an issue that `0` and `never` behave identically, then I would rather change the type to `types.ints.positive` and require `never` for disabling sleep.

Reply:
```suggestion
      type = types.nullOr (types.either types.ints.positive (types.enum ["never"]));
```

Sounds good to me 👍

Reply:
Done